### DISCUSSION TITLE
Invert green channel of Normal map textures.

### DISCUSSION THREAD

**Discussion by arun54321** [ImageMagick/ImageMagick#2118] [Help] [answered]
I want to convert several DirectX normal maps to openGL normal maps by inverting thier green channel. How can I do that using imagemagick or mogrify?

**Comment by fmw42** [comment, marked as answer]
I do not know if ImageMagick can read DirectX or write openGL as I do not use them. But you can invert the green channel as follows for ImageMagick 6.x.x.x

`convert input -channel green -negate +channel output`

Please always provide your ImageMagick version and platform when asking questions on this forum as syntax may vary.  If using ImageMagick 7.x.x.x, then replace convert with magick.

**Reply by arun54321** [reply]
They are regular png and jpeg files. Your solution works. Thank you.
